File: lib-database/src/main/java/com/ustadmobile/core/db/dao/StateContentDao.java
package com.ustadmobile.core.db.dao;

import com.ustadmobile.lib.database.annotation.UmDao;
import com.ustadmobile.lib.database.annotation.UmQuery;
import com.ustadmobile.lib.database.annotation.UmRepository;
import com.ustadmobile.lib.db.entities.StateContentEntity;
import com.ustadmobile.lib.db.sync.dao.SyncableDao;

import java.util.List;

@UmDao
@UmRepository
public abstract class StateContentDao implements SyncableDao<StateContentEntity, StateContentDao> {

    @UmQuery("SELECT * FROM StateContentEntity WHERE stateContentStateUid = :id AND isactive")
    public abstract List<StateContentEntity> findAllStateContentWithStateUid(long id);

    @UmQuery("SELECT * FROM StateContentEntity WHERE stateContentStateUid = :stateUid AND stateContentKey = :key AND isactive")
    public abstract StateContentEntity findStateContentByKeyAndStateUid(String key, long stateUid);

    @UmQuery("UPDATE StateContentEntity SET isactive = :isActive WHERE stateContentUid = :stateUid")
    public abstract void setInActiveStateContentByKeyAndUid(boolean isActive, long stateUid);


}

File: lib-database/src/commonMain/kotlin/com/ustadmobile/core/db/dao/StateContentDao.kt
package com.ustadmobile.core.db.dao

import androidx.room.Query
import com.ustadmobile.door.annotation.*
import com.ustadmobile.lib.db.entities.StateContentEntity

@DoorDao
@Repository
expect abstract class StateContentDao : BaseDao<StateContentEntity> {

    @Query("SELECT * FROM StateContentEntity WHERE stateContentStateUid = :id AND isIsactive")
    abstract fun findAllStateContentWithStateUid(id: Long): List<StateContentEntity>

    @Query("SELECT * FROM StateContentEntity WHERE stateContentStateUid = :stateUid AND stateContentKey = :key AND isIsactive")
    abstract fun findStateContentByKeyAndStateUid(key: String, stateUid: Long): StateContentEntity?

    @Query("""
        UPDATE StateContentEntity 
           SET isIsactive = :isActive,  
               stateContentLct = :updateTime
         WHERE stateContentUid = :stateUid
    """)
    abstract fun setInActiveStateContentByKeyAndUid(
        isActive: Boolean,
        stateUid: Long,
        updateTime: Long,
    )


}
